### PR TITLE
[WOR-523] Add landing zone-related fields to Rawls billing project record

### PR DESCRIPTION
**Pull request by ghost**
Ticket: [WOR-523](https://broadworkbench.atlassian.net/browse/WOR-523)
- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check that the **Product Owner** has signed off on any user-facing changes
- [ ] **Submitter**: Make sure Swagger is updated if API changes
  - [ ] **...and Orchestration's Swagger too!**
- [ ] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Database checks:
  * If PR includes new or changed db queries, include the explain plans in the description
  * Make sure liquibase is updated if appropriate
  * If doing a migration, take a backup of the
  [dev](https://console.cloud.google.com/sql/instances/terraform-qfarbdq3lrexxck5htofjs5z6m/backups?project=broad-dsde-dev&organizationId=548622027621)
  and
  [alpha](https://console.cloud.google.com/sql/instances/terraform-r4caezzc35c4tb7pgdhwkmme4y/backups?project=broad-dsde-alpha&organizationId=548622027621)
  DBs in Google Cloud Console
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] Tell your tech lead (TL) that the PR exists if they want to look at it
- [ ] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via Slack and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: core/src/main/resources/org/broadinstitute/dsde/rawls/liquibase/changesets/20220929_creating_landing_zones.xml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<databaseChangeLog logicalFilePath="dummy" xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+                   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                   xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.4.xsd">
+    <changeSet logicalFilePath="dummy" author="yunip20" id="create_CREATING_LANDING_ZONES">
+        <createTable tableName="CREATING_LANDING_ZONES">
```
Comment (cahrens) [inline core/src/main/resources/org/broadinstitute/dsde/rawls/liquibase/changesets/20220929_creating_landing_zones.xml]: As we discussed in refinement, I think we want to generalize this table so it is a workspace manager resource monitor. The primary key would then be the resource ID, the job control ID shouldn't mention landing zones specifically, and we want an entry for the resource type.

File: core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/RawlsBillingProjectComponent.scala [hunk 2]
```diff
@@ -99,6 +99,13 @@ final case class BillingAccountChange(id: Long,
                                       outcome: Option[Outcome]
 )
 
+final case class WorkspaceManagerResourceMonitor(jobControlId: UUID,
```
Comment (cahrens) [inline core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/RawlsBillingProjectComponent.scala]: I would find it clearer if this was named `WorkspaceManagerResourceMonitorRecord`.

File: core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/RawlsBillingProjectComponent.scala [hunk 3]
```diff
@@ -144,6 +151,25 @@ trait RawlsBillingProjectComponent {
...
+    def billingProjectId = column[Option[String]]("BILLING_PROJECT_ID")
+
+    def createdTime = column[Instant]("CREATED_TIME")
+
+    def * = (jobControlId,
+      jobType,
+      workspaceId,
+      billingProjectId,
+      createdTime
+    ) <> ((WorkspaceManagerResourceMonitor.apply _).tupled, WorkspaceManagerResourceMonitor.unapply)
```
Comment (cahrens) [inline core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/RawlsBillingProjectComponent.scala]: @marctalbott do we need to add anything in the model for the foreign keys/referenced tables (I'm not that familiar with slick)?

File: core/src/main/resources/org/broadinstitute/dsde/rawls/liquibase/changesets/20220929_create_workspace_manager_resource_monitor.xml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+                <constraints nullable="true" referencedTableName="WORKSPACE" referencedColumnNames="id" foreignKeyName="FK_WORKSPACE_ID" deleteCascade="true"/>
+            </column>
+            <column name="BILLING_PROJECT_ID" type="VARCHAR(254)">
+                <constraints nullable="true" referencedTableName="BILLING_PROJECT" referencedColumnNames="NAME" foreignKeyName="FK_BILLING_PROJECT_ID" deleteCascade="true"/>
+            </column>
+            <column name="CREATED_TIME" type="DATETIME(6)" defaultValueComputed="NOW(6)">
+                <constraints nullable="false"/>
+            </column>
+        </createTable>
+    </changeSet>
```
Comment (marctalbott) [inline core/src/main/resources/org/broadinstitute/dsde/rawls/liquibase/changesets/20220929_create_workspace_manager_resource_monitor.xml]: We also need to add the nullable UUID `LANDING_ZONE_ID` to the `BILLING_PROJECT` table

File: core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/RawlsBillingProjectComponent.scala (hunk 3 above)
Comment (blakery) [reply core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/RawlsBillingProjectComponent.scala]: Yes: here's the [documentation for foreign keys](https://scala-slick.org/doc/3.0.0/schemas.html#constraints)

Comment (marctalbott) [reply core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/RawlsBillingProjectComponent.scala]: Hmm, looks like Slick does provide [additional FK support](https://scala-slick.org/doc/3.0.0/schemas.html#constraints), but we don't use it elsewhere in Rawls afaik and it doesn't appear to provide many benefits in addition to the FK constraint defined in the actual DB. I think this is fine as it is

**Review by marctalbott — APPROVED** [review]
Looks good!

**Review by blakery — APPROVED** [review]

**Comment by blakery** [comment]
If you're having issues with the Jenkins Automation Tests, you may need to merge(or rebase) from develop. Those tests are often updated based on changes to the project, and they are in a separate repo (not versioned with rawls), so sometimes you need changes that have made it to develop in order for them to pass.

**Pull request merged**